src/client/contextChannels.ts

```diff
@@ -43,6 +45,10 @@ export interface ChannelChangedEvent {
     previousChannel: Channel|null;
 }
 
+interface ChannelContextListener extends ContextListener {
+    id: ChannelId;
```

Comment:
Are you saying let's have a full-fat channel object here rather than the id? This only ever gets used for checking against the id of incoming objects, and this interface isn't exposed anywhere so we can change it later if there's need to